README.md

```diff
@@ -1,73 +1,320 @@
+
+### hosts
+Each key in this dictionary is the unique name of a host. For each host key in this dictionary, the following host-specific parameters can be specified.  
+ 
+| Parameter            | Description                                                                                       | Type        | Required | Default                 | Libreswan Equivalent |
+|----------------------|---------------------------------------------------------------------------------------------------|:-----------:|:--------:|-------------------------|:--------------------:|
+| host                 | A host/IP to use for setting up a vpn connection. Only specified when overriding the name of hosts used by Ansible for SSH. | str | no   | - | left/right           |
+| public_key_src       | A path to a file on the controller host containing the public key used by this host for public key authentication without certificates.     | str    | no       | See [public_key fields](#public_key)| leftrsasigkey/rightrsasigkey |
+| public_key_content   | Contains the public key used by this host for public key authentication without certificates.                        | str         | no       | -                       | leftrsasigkey/rightrsasigkey |
+| cert_name            | Defines the certificate nickname of this host's certificate in the NSS database.                  | str         | no       | -                       | leftcert/rightcert   |
+| external             | Indicates that the associated host is external to the inventory (e.g. not managed by our inventory).    | bool        | no       | False                   | -                    |
+| subnets              | Defines a list of the subnets that should be available via the VPN connection.                    | list        | no       | -                       | leftsubnets/rightsubnets |
+
+#### public_key
+
+`public_key_src` specifies a path to a file on the controller host containing the public key used by this host for public key authentication without certificates. Otherwise, the user can directly specify the public key for this host by populating `public_key_content`. `public_key_content` can also accept a CKAID or nickname for a public key in the NSS database.
+
+Note that `public_key_src` and `public_key_content` may also be specified as host variables in the Ansible `all.hosts` section. The variable names in this case will be `vpn_public_key_src` and `vpn_public_key_content`. See [this example](#defining-public-keys-at-different-levels).
+
+**To be added later**: if neither `public_key_src` nor `public_key_content` is populated, the role will generate key pairs for each host.
+
+### Handling special cases with host names
+
+There are two ways to handle the special case when the names of the hosts in the Ansible inventory are not FQDN hostnames that you can use directly (e.g. they may be host aliases like "bastion_east"):  
+
+1. **Use the Ansible `hosts` configuration**
```

Comment:
This seems strange, first of all it is not entirely clear what is "the Ansible `hosts` configuration" (what is intended is probably the Ansible's list of hosts, i.e. the inventory), and second, you are using the inventory in any case, since above you wrote "This assumes the names of the hosts under `hosts` are the same as the names of the hosts used in the Ansible inventory". What is really specific to this part is the use of the `ansible_host` variable.